### DISCUSSION TITLE
WPF Magic NavigationBar (ListBox)

### DISCUSSION THREAD

**Discussion by vickyqu115** [dotnet/wpf#8704] [Show and tell]
Hello, my name is Vicky. I am a Chinese developer and video creator active in South Korea.

I am excited to introduce the Magic NavigationBar control, a CustomControl created by inheriting from the WPF ListBox. The original source code along with a demo project illustrating its usage is available on the GitHub repository.

To share my technical knowledge with a wider audience, I have made my content available on GitHub, YouTube, BiliBili, and CodeProject. My goal is to release one piece of content every week (although it's quite challenging).

I hope you find it interesting and enjoyable. Thank you.

[CodeProject](https://www.codeproject.com/Articles/5375482/Customizing-ListBox-for-a-Smooth-Animated-Navigati)
[Youtube Tutorial](https://youtu.be/dxuLWlukthg)
[BiliBili Tutorial](https://www.bilibili.com/video/BV1Ui4y1a717)
[GitHub Repository](https://github.com/vickyqu115/navigationbar)

### NavigationBar
> WPF applications traditionally prefer a programmatic approach that connects multiple screens through menu configurations and presents them in a unified manner. This technique, often referred to as the menu or Navigation, is one of the core implementations in WPF. It also has a direct correlation with the architecture (design) of the project, so paying more attention to its implementation can positively impact the quality of the project.
>
> This control features a design and animations specialized for mobile, but it can be elegantly and structurally implemented using ListBox and Animation technologies available in WPF. Additionally, it can be similarly implemented in Cross-Platform environments such as AvaloniaUI, Uno, OpenSilver, MAUI, which allows this project to be researched and applied across various platforms.
>
> The goal is also to widely promote the flexibility and excellence of WPF implementation and share the technology. Through this project, we hope you will deeply experience the charm of WPF.

![295754446-7b360df7-b62b-440b-a43d-d5b4926d4b7b](https://github.com/dotnet/wpf/assets/101777355/d4470404-dc50-4acd-96fe-f3d084fa401b)

https://youtu.be/dxuLWlukthg

**Comment by lindexi** [comment]
Awesome. What a cool effect!
